src/components/ip-checkbox-list/ip-checkbox-list.tsx

```diff
@@ -0,0 +1,67 @@
+import { Component, Prop, h, State, Watch, EventEmitter, Event } from '@stencil/core';
+
+interface CheckboxOption {
+  id: string;
+  label: string;
+}
+
+@Component({
+  tag: 'ip-checkbox-list',
+  styleUrl: 'ip-checkbox-list.scss',
+  shadow: true,
+})
+export class IpCheckboxList {
+  @Prop() options: string;
+  @State() selectedOptions: string[] = [];
+
+  parsedOptions: CheckboxOption[] = [];
+
+  @Event({ bubbles: true, composed: true }) selectionChanged: EventEmitter<string[]>;
+
+  componentWillLoad() {
+    this.parseOptions(this.options);
+  }
+
+  @Watch('options')
+  parseOptions(newValue: string) {
+    try {
+      this.parsedOptions = JSON.parse(newValue);
+    } catch (error) {
+      console.error('Invalid options:', error);
+    }
+  }
+
+  handleChange(optionId: string) {
+    const index = this.selectedOptions.indexOf(optionId);
+
+    if (index < 0) {
+      this.selectedOptions = [...this.selectedOptions, optionId];
+    } else {
+      this.selectedOptions = [...this.selectedOptions.slice(0, index), ...this.selectedOptions.slice(index + 1)];
+    }
```

Comment:
I think using .push() method to add element and .filter() to remove is more readable

Reply:
**push()** mutate existing but **filter()** return the result array after filter

Reply:
Yep miss the filter part of you comment. 🙏

Is filter keep existing array index or reorder from 0 to N ?

In other langages got multiple times confusions because of unsequential index after filter. But forgot the js behavior

Reply:
filter does not mutate existing array, it returns a new array with the filtered value therefore it keeps existing array index.